service/matching/context.go

```diff
@@ -92,6 +92,17 @@ func (reqCtx *handlerContext) handleErr(err error) error {
 
 	scope := reqCtx.scope
 
+	switch err {
+	case context.DeadlineExceeded:
+		scope.IncCounter(metrics.ServiceErrContextTimeoutCounter)
+		return serviceerror.NewDeadlineExceeded(err.Error())
+	case context.Canceled:
+		scope.IncCounter(metrics.ServiceErrContextTimeoutCounter)
+		return serviceerror.NewCanceled(err.Error())
+	default:
+		// noop
+	}
```

Comment:
Can you use the same approach as workflow handler here:
```suggestion
	if common.IsContextDeadlineExceededErr(err) || common.IsContextCanceledErr(err) {
		scope.IncCounter(metrics.ServiceErrContextTimeoutCounter)
		return err
	}
```

and let `serviceerror.ToStatus` convert errors.